dcnum/feat/feat_moments/mt_legacy.py

```diff
@@ -11,67 +11,99 @@ def moments_based_features(mask, pixel_size):
     size = mask.shape[0]
 
     empty = np.full(size, np.nan, dtype=np.float64)
-    deform = np.copy(empty)
+
+    # features from raw contour
     size_x = np.copy(empty)
     size_y = np.copy(empty)
-    pos_x = np.copy(empty)
-    pos_y = np.copy(empty)
+    aspect = np.copy(empty)
     area_msd = np.copy(empty)
+    area_um_raw = np.copy(empty)
+    per_um_raw = np.copy(empty)
+    deform_raw = np.copy(empty)
     area_ratio = np.copy(empty)
-    area_um = np.copy(empty)
-    aspect = np.copy(empty)
-    tilt = np.copy(empty)
-    inert_ratio_cvx = np.copy(empty)
+    per_ratio = np.copy(empty)
     inert_ratio_raw = np.copy(empty)
     inert_ratio_prnc = np.copy(empty)
+    eccentr_prnc = np.copy(empty)
+    tilt = np.copy(empty)
+    s_x = np.copy(empty)
+    s_y = np.copy(empty)
+
+    # features from convex hull
+    pos_x = np.copy(empty)
+    pos_y = np.copy(empty)
+    area_um = np.copy(empty)
+    deform = np.copy(empty)
+    inert_ratio_cvx = np.copy(empty)
     # The following valid-array is not a real feature, but only
     # used to figure out which events need to be removed due
     # to invalid computed features, often due to invalid contours.
     valid = np.full(size, False)
 
     for ii in range(size):
-        cont_raw = contour_single_opencv(mask[ii])
+        # raw contour
+        cont_raw = contour_single_opencv(mask[ii]).astype("float32")
         if len(cont_raw.shape) < 2:
             continue
         if cv2.contourArea(cont_raw) == 0:
             continue
+
         mu_raw = cv2.moments(cont_raw)
+        arc_raw = np.float64(cv2.arcLength(cont_raw, True))
+        area_raw = np.float64(mu_raw["m00"])
+        event_mask = mask[ii]
 
         # convex hull
-        cont_cvx = np.squeeze(cv2.convexHull(cont_raw))
+        cont_cvx = np.squeeze(np.int64(cv2.convexHull(cont_raw)))
         mu_cvx = cv2.moments(cont_cvx)
+        arc_hull = np.float64(cv2.arcLength(cont_cvx, True))
+        # circ
+        circ = 2 * np.sqrt(np.pi * mu_cvx["m00"]) / arc_hull
+        deform[ii] = 1 - circ
 
         if mu_cvx["m00"] == 0 or mu_raw["m00"] == 0:
-            # Contour size too small
+            # contour size too small
             continue
 
-        arc = cv2.arcLength(cont_cvx, True)
+        # moments of inetria
+        i_xx = np.float64(mu_raw["mu02"])
+        i_yy = np.float64(mu_raw["mu20"])
+        i_xy = np.float64(mu_raw["mu11"])
+        # central moments in principal axes
+        i_1 = 0.5*(i_xx+i_yy + np.sqrt((i_xx - i_yy) ** 2 + 4*(i_xy ** 2)))
+        i_2 = 0.5*(i_xx+i_yy - np.sqrt((i_xx - i_yy) ** 2 + 4*(i_xy ** 2)))
 
+        # bounding box
         x, y, w, h = cv2.boundingRect(cont_raw)
 
-        # tilt
```

Comment:
You have refactored the tilt computation quite a bit. Have you checked that the new computation yields the exact same result? Should we save this for a separate PR?

Reply:
yes, both tilt and inert_ratio_prnc are the same with the new computation. You can also see that the tests past. This is what I gave you back then as a code but did not have time to add it.
I really don´t have time to create new PR and i don´t see a benefit of it. Can you, please add that this is new cell features computation together with improving calculation of tilt and inert_ratio_prnc.

Reply:
The benefit of a new PR is that each change is properly tracked. I am planning to squash-merge this PR, and changes in so many places make it hard to find commits that introduced regressions later on. Please take the time to create a new PR. This is important.